File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int maxi = 0;
        int n = prices.size();

        for(int i = n - 1; i >= 0; i--){
            maxi = max(maxi, prices[i]);
            ans = max(ans, maxi - prices[i]);
        }

        return ans;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        current_maxi = prices[len(prices) - 1]
        result = 0

        for i in range(len(prices) - 1, -1, -1):
            current_maxi = max(current_maxi, prices[i])
            result = max(result, current_maxi - prices[i])

        return result
